packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useCallback, useState } from 'react'
+import Form from '@rjsf/mui'
+import validator from '@rjsf/validator-ajv8'
+import { JSONSchema7 } from 'json-schema'
+import ArrayFieldDescriptionTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldDescriptionTemplate'
+import ArrayFieldItemTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldItemTemplate'
+import ArrayFieldTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldTemplate'
+import ArrayFieldTitleTemplate from '../SchemaDrivenAnnotationEditor/template/ArrayFieldTitleTemplate'
+import ButtonTemplate from '../SchemaDrivenAnnotationEditor/template/ButtonTemplate'
+import { Box, Button, Collapse, TextField } from '@mui/material'
+import { parse } from 'papaparse'
+type JSONArrayEditorProps = {
+  value: string[]
+  onChange: (newValue: string[]) => void
+  onSubmit: (formData: string[]) => void
+}
+
+const arraySchema: JSONSchema7 = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'array',
+  items: {
+    type: 'string',
+  },
+}
+
+const JSONArrayEditor = React.forwardRef(function JSONArrayEditor(
+  props: JSONArrayEditorProps,
+  ref: React.Ref<Form>,
+) {
+  const { value, onChange, onSubmit } = props
+  const [showPasteNewValuesForm, setShowPasteNewValuesForm] = useState(false)
+  const [pastedValues, setPastedValues] = useState('')
+
+  const addPastedValuesToArray = useCallback(() => {
+    if (pastedValues) {
+      parse<string[]>(pastedValues, {
+        complete: result => {
+          console.log(result)
+          if (result.errors.length > 0) {
+            console.error('Error parsing pasted values', result.errors)
+          }
+          onChange([...value, ...result.data[0]])
+          setPastedValues('')
+          setShowPasteNewValuesForm(false)
+        },
+      })
+    } else {
+      setPastedValues('')
+      setShowPasteNewValuesForm(false)
+    }
+  }, [onChange, pastedValues, value])
+
+  return (
+    <Box
+      className="AnnotationEditor"
+      sx={{
+        // Hide the label/button to show more info
+        '.LabelContainer': {
+          display: 'none',
+          visibility: 'hidden',
+        },
+      }}
+    >
+      <Form
+        ref={ref}
+        schema={arraySchema}
+        uiSchema={{
+          'ui:submitButtonOptions': {
+            norender: true,
+          },
+        }}
+        validator={validator}
+        formData={value}
+        onChange={({ formData }) => onChange(formData)}
+        onSubmit={({ formData }) => onSubmit(formData)}
+        templates={{
+          ArrayFieldDescriptionTemplate: ArrayFieldDescriptionTemplate,
+          ArrayFieldItemTemplate: ArrayFieldItemTemplate,
+          ArrayFieldTemplate: ArrayFieldTemplate,
+          ArrayFieldTitleTemplate: ArrayFieldTitleTemplate,
+          ButtonTemplates: ButtonTemplate,
```

Comment:
More work could be done to separate Annotations-specific functionality across all of our RJSF custom components, but this set seems to work for this purpose.